Add `Price Freeze Period` section. (#1478)
saleor/saleor-docs: docs/developer/price-expiration.mdx

---
title: Price Expiration
sidebar_label: Price Expiration
---

## Introduction
When adding a line to an order, the product variant price is taken from the relevant channel listing. 
However, throughout the order lifecycle, price recalculations may be required, and channel listing prices can be updated in the meantime.
To ensure price consistency over time, regardless of channel listing updates, Saleor provides a settings that allows to control the price freeze period.

### Settings
You can configure the duration after which a price is considered outdated and requires refreshing from the channel listings. 
This is managed through the [`channel`](developer/channels/configuration.mdx) settings:
- `draftOrderLinePriceFreezePeriod`:  Defines the time (in hours) after which the order line price will be refreshed. The default value is 24 hours. 
To prevent price refreshing entirely, set this value to null or 0.

> - `checkoutLinePriceFreezePeriod`: To be added. Currently, each recalculation retrieves the latest price from the channel listing.

Reviewer: This could be misleading, as you provide the `checkoutLinePriceFreezePeriod`, which does not exist. Personally, I would add a separate task about updating this section when we introduce the same flow for the checkout. 
But I would keep the sentence about checkout part: 

> Currently, each recalculation retrieves the latest price from the channel listing.

Also IMHO, would be goot to mention that this flow was added in 3.21

---
title: Price Expiration
sidebar_label: Price Expiration
---

:::info
This feature was introduced in **Saleor 3.21**. In previous versions, the base prices were refreshed with every single recalculation.
:::

## Introduction
When adding a line to an order, the product variant price is taken from the relevant channel listing. 
However, throughout the order lifecycle, price recalculations may be required, and channel listing prices can be updated in the meantime.

> To ensure price consistency over time, regardless of channel listing updates, Saleor provides a settings that allows to control the price freeze period.

Reviewer 2: I have a trouble to understand what do you want to tell us.
`price recalculations may be required` - because of what?
`and channel listing prices can be updated in the meantime.` - how it's related with other part of the sentence?

> To ensure price consistency over time, regardless of channel listing updates, Saleor provides a settings that allows to control the price freeze period.

I don't think that mentioning here channel listings update is important. 
I think that introduction should say what happened in general, not going into the details

Author: > I don't think that mentioning here channel listings update is important.

This is the essence of this change - we don't fetch prices from channel listing, but we use denormalized ones.

> price recalculations may be required - because of what?

 As you said, this is introduction so it shouldn't got much into details what can trigger the price recalculation

### Settings

> You can configure the duration after which a price is considered outdated and requires refreshing from the channel listings. 

Reviewer 2: Maybe:
```suggestion
You can configure the time period after which a price is considered outdated and needs to be refreshed from the channel listings.
```

> This is managed through the [`channel`](developer/channels/configuration.mdx) setting `draftOrderLinePriceFreezePeriod`, 
> which defines the time in hours after which the order line price will be refreshed. The default value is 24 hours. 

Reviewer 2: ```suggestion
This is controlled by the [`channel`](developer/channels/configuration.mdx) setting `draftOrderLinePriceFreezePeriod`. It defines how long the price remains fixed before being refreshed, measured in hours. The default setting is 24 hours.
```

Author: corrected

> To prevent price refreshing entirely, set this value to `null` or `0`.

Reviewer 2: ```suggestion
To disable automatic price refreshing, set this value to `null` or `0`.
```

Author: corrected

### How It Works

> The price expiration mechanism applies only to orders with `DRAFT` status. Orders in any other status will retain their original prices permanently.

Reviewer: I would suggest three separate sub-sections. First for `DRAFT`, second  for `UNCONFIRMED` and the last that would explicitly say that we don't modify the prices.

Right now, you have a section `###Settings`. You explain there that you can control the expiration time for orders.
Later in section `How it works` you're pointing that the price expiration works only for DRAFT. So for someone who is not familiar this can be hard to understand (for example, does the `settings` part is related to ALL orders? or related to only specific orders? We don't say that explicit).

So IMHO, to keep it clear, I would expect the section for DRAFT, with the info that you already provided.
Second for UNCONFIRMED, where you say: we don't update the prices for existing order's content, but we will use the actual price for channel when you add the new line
Thrid one would just say: order is confirmed so we don't make any calculations.

> A new price expiration time is set on an order line when:

Reviewer 2: Do not use `new` - how long it's gonna be `new`? Will you remember to change it after it won't being `new`? 😄 
```suggestion
The price expiration time is set on an order line when:
```

Author: corrected

- A variant is added to the order.

> - The freeze period has passed, and the line undergoes recalculation.

Reviewer 2: `and the line undergoes recalculation.` - what do you mean exactly?

Author: corrected

> Once the price freeze period expires, Saleor retrieves the latest variant price from the channel listing and updates it accordingly. 
> This update also includes any changes in applied discounts. 
> For example, if a voucher reduced the line price and the voucher reward value has changed since the initial calculation, the line price will be adjusted based on the new reward value.

Reviewer 2: ```suggestion
Once the price freeze period expires, Saleor retrieves the latest variant price from the channel listing and recalculates the order line price. This update also reflects any changes in applied discounts.
For example, if a voucher initially reduced the line price but its value has changed since the initial calculation, the line price will be adjusted accordingly.
```

Author: > For example, if a voucher initially reduced the line price but its value has changed since the initial calculation, the line price will be adjusted accordingly.

With suggested changes, I don't get what `its value` refers to and what `the line price will be adjusted accordingly` means, so corrected partially.
